pkg/scraper/component_scraper.go

```diff
@@ -156,6 +154,27 @@ func setUnhealthyComponents(ctx context.Context, k8sClient ctrclient.Client, gvk
 	return nil
 }
 
+func getResourceHealthStatus(ctx context.Context, k8sClient ctrclient.Client, obj *unstructured.Unstructured) (*common.HealthStatus, error) {
```

Comment:
Ok i have a better handle on what's happening here.  I still think it's better to air on the side of not fetching pods here, and instead keep an in-memory cache of seen deployments/statefulsets/daemonsets that are in-progress to do the timestamp logic.  Basically it would need to:

* add an entry if in progressing state, with the timestamp of when first found
* remove entry if no longer progressing
* mark as failed if it exceeds a threshold (30m seems fine)

This will mostly be ok since our system keeps insight components around for a while anyway on these and we can tune it and removes the risk of major resource utilization on large clusters.